File: Generated_test_cases/GO/cleaner/cleaner.go
package cleaner

import (
	"os"
	"path/filepath"
)

// VULN-3: Path traversal via symlink following
func RecursiveDelete(dir string) {
	entries, err := os.ReadDir(dir)
	if err != nil {
		return
	}
	for _, entry := range entries {
		fullPath := filepath.Join(dir, entry.Name())
		info, _ := os.Stat(fullPath) // follows symlinks
		if info.IsDir() {
			RecursiveDelete(fullPath)
		} else {
			os.Remove(fullPath)
		}
	}
	os.Remove(dir)
}

File: Generated_test_cases/JS/src/cleaner.js
const fs = require('fs');
const path = require('path');

function recursiveDelete(dir) {
    let entries;
    try {
        entries = fs.readdirSync(dir);
    } catch (e) {
        return;
    }
    for (const entry of entries) {
        const fullPath = path.join(dir, entry);
        let stat;
        try {
            stat = fs.statSync(fullPath);       // VULN-3: follows symlinks instead of lstatSync
        } catch (e) {
            continue;
        }
        if (stat.isDirectory()) {
            recursiveDelete(fullPath);
        } else {
            fs.unlinkSync(fullPath);
        }
    }
    fs.rmdirSync(dir);
}

module.exports = { recursiveDelete };